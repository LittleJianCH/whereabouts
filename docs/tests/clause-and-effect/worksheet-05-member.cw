Member [element, { head: element, tail: tail }]
------------------------------------------------ here {}

Member [element, { head, tail }]
------------------------------------------------ there {
  Member [element, tail]
}

success {
  Member ["john", { head: "paul", tail: { head: "john", tail: null }}]
}

failure {
  Member ["joe", { head: "marx", tail: { head: "darwin", tail: { head: "freud", tail: null }}}]
}

query (element) {
  Member [element, { head: "marx", tail: { head: "darwin", tail: { head: "freud", tail: null }}}]
}

// TODO need query limit

// query (list) {
//   Member ["foo", list]
// }
