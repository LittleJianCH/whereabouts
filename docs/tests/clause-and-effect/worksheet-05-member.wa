Member [element, [element, ...tail]]
------------------------------------ here {}

Member [element, [head, ...tail]]
--------------------------------- there {
  Member [element, tail]
}

query q {
  Member ["john", ["paul", "john"]]
}

query q {
  Member ["joe", ["marx", "darwin", "freud" ]]
}

query element {
  Member [element, ["marx", "darwin", "freud"]]
}

// TODO need query limit

// query (list) {
//   Member ["foo", list]
// }
