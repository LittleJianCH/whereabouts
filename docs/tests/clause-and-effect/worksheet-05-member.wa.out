{ "success": true, "count": 3, "solutions": [["marx"], ["darwin"], ["freud"]] }
